### DISCUSSION TITLE
"No Response" after running 05-WorkingLED accessory for several minutes

### DISCUSSION THREAD

**Discussion by sweetsmarthome** [HomeSpan/HomeSpan#334] [Q&A] [answered]
I have a problem that my 05-WorkingLED example accessory eventually turns to “No Response” in the Home.app window after normal operation for around 3 to 30 minutes. It takes a long time (usually more than 10 minutes) to resume responding.
![noresponse](https://user-images.githubusercontent.com/108333233/176589022-9d91716d-ee99-4650-9b14-3b5fe33153d8.jpg)
As all other accessories including Meross plugs and Homebridge work fine, I think this is not a HomeKit or WiFi problem but an ESP32 problem. Arduino serial monitor still works in the "No Response" situation, and it says that all the HAP Connections in slots #0 to #13 are unconnected. The “R” command, which invokes ESP.restart(), make the accessory work again in the Home.app. I have managed this problem by monitoring HAP accessory slots and invoking ESP.restart() when all the HAP slots are unconnected. 

Apparently, this is not a smart solution. Is there any better way to solve this kind of “No Response” problem?

**Comment by HomeSpan** [comment]
I received an email that you posted an answer to your quest above related to the version of he ESP32 chip you are using (Rev 1 vs Rev 3).  However, that answer is no longer posted - did you find it was not in fact related to Rev 1 vs Rev 3?

**Reply by sweetsmarthome** [reply]
Yes, rev. 3 seemed to be fine when I posted, but after a half day operation, it also turned to No-Response. Although rev. 3 looks a little better, there is no significant difference between rev. 1 and 3, so that I have canceled my post.

**Comment by HomeSpan** [comment]
Also, are you using the latest version of the Arduino-ESP32 board manager (version 2.0.3?).  There was a problem with the UDP/MDNS function in earlier versions that I believe Espressif fixed in version 2.0.1, which virtually eliminated the issue (when the problem was caused by MDNS messages that were rejected by certain routers).

**Reply by sweetsmarthome** [reply]
I am using the Arduino-ESP Ver. 2.0.3.

**Reply by blaisebarrette** [reply]
I don't know if this can be of any help but I had a very stable connection for a good while and for no apparent reason I started to get the dreadful 'No response' message. After trying all sorts of thing, I finally discovered that I have three HomeKit hubs in my house, Two AppleTVs and a HomePod mini. The active hub was the AppleTV in the master bedroom witch is an older version, connected via wifi and in the furthest room from the router. I deactivated it as a hub and made sure the Living-room Apple TV that is connected via ethernet and is in a more central location was the active HomeKit hub. Since then, it's been rock steady!

**Reply by sweetsmarthome** [reply]
Thank you for the info.
I also have two home hubs, a HomePod mini and an old AppleTV HD. Seeing the connection list shown by 's' command in the Arduino serial monitor, the HomePod mini is the active (admin) hub. I have tried to delete my Apple ID set in the AppleTV, to connect it via Ethernet, and finally to power off the AppleTV, however, the HomePod is lost in the connection list and the EPS32 accessory goes to 'No response' after 5-6 minutes operation. Other HAP accessories such as a Meross smart plug continue responding as usual. 

**Edit**: After leaving the old AppleTV HD disconnected for half a day, the connection seems to be improved. I will continue checking for a while.  

**Edit 2**: The ESP32 is still responding after one and half a day (36 hours). I will re-connect the Apple TV to see the cause and effect.

**Comment by blaisebarrette** [comment]
You can access the HomePod mini via the Home app and restart it. Maybe momentarily disconnect the power cord from the Apple TV before you restart the HomePod mini. Just to make sur the HomePod mini will have no choice to connect as primary hub.

**Comment by sweetsmarthome** [comment, marked as answer]
My ESP32 accessory went to "No Response" again after normal operation of several days. Now I have turned off my old Apple TV, however "No Response" happens after around 10 min operation. I will continue testing using simplest 01-SimpleLightBulb. Apple TV and name conflict may not be the real cause. Sorry for my confusing report. 

-----------------------------

Solved!

Finally I think I have fixed the "No Response" problem (so far). I have accidentally assigned same name ("Living Room") to both the room and the Apple TV, and this likely causes the No Response problem of my HomeSpan accessory. 

I set up my old Apple TV HD (4th gen.) long ago when I was not using HomeKit. At that time, the set-up program suggested me to name the Apple TV after the place where it is installed. I selected "Living Room" from the examples shown on the screen. Later, I installed HomeKit and named the room "Living Room" also by selecting one of the options shown in the Home.app program. As a result, the Apple TV and the room shared identical name "Living Room". I noticed that this may cause a problem when Apple TV suggested me to rename it to "Living Room (2)" in its HomeKit re-connection menu. I have re-named the Apple TV to totally different name, "Apple TV HD", then the "No Response" problem never happens so far.

Actually, assigning same name both to a room and an accessory is not allowed in Home.app operations. Following screen shot shows a message when I tried to re-name the Apple TV back to "Living Room". I thinks this implies that name conflict may cause a problem in HomeKit protocol.

![image](https://user-images.githubusercontent.com/108333233/181436530-88ed38cf-d2ba-4911-acf0-8e67c226b167.png)


**Summary**: When "No Response" happens frequently, checking name conflict among items, such as old Apple TV, may help. 

Thank you [HomeSpan](https://github.com/HomeSpan) and [blaisebarrette](https://github.com/blaisebarrette).

**Reply by HomeSpan** [reply]
That's very good news, though it's puzzling this only impacted HomeSpan and not HomeBridge devices, unless HomeBridge has some built-in protection (e.g. restarting if all connections are dropped).

**Reply by sweetsmarthome** [reply]
Yes, it is puzzling. Merros Smart Plug, the only Apple approved HAP accessory I have, was also not affected. The plug may have some protection.

I tried to disconnect and re-connect the Apple TV to HomeKit again, and  found that assigning-name-menu was disappeared from the Apple TV menu. It is automatically named as the product name "Apple TV" just like my HomePod mini. This kind of trouble may only happen to specific old Apple TV users who are using outdated configuration made by old version of tvOS. I think no protection is required in HomeSpan, as it may affect responsibility in normal use.

My HomeSpan program is working without problem since re-configuring the Apple TV. Thank you very much for providing excellent toolkit.
